src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+    private final Map<String, Integer> storage = new HashMap<>();
```

Comment:
According to the encapsulation principle, the storage map should be private. It's good that you have used private access modifier here, but ensure that the returned storage map cannot be modified directly from outside the class. This could be done by returning a copy of the map or an unmodifiable view.